src/pages/orders/index.tsx

```diff
@@ -0,0 +1,141 @@
+import { useState, useMemo, useCallback } from "react";
+import { Heading, InputGroup, Input, InputLeftElement, Flex, Box } from "@chakra-ui/react";
+import { AiOutlineSearch } from "react-icons/ai";
+import { useGetAllOrdersQuery } from "../../redux/api/ordersApi";
+import OrderContainer from "./components/OrderContainer";
+import OrderStatusDropdownFilter from "./components/OrderStatusDropdownFilter";
+import PaginationButton from "@/components/PaginationButton.tsx";
+import { LIMIT, OFFSET } from "@/constants/paginationData";
+import { FilterType } from "./components/OrderStatusDropdownFilter";
+import { IOrder } from "@/interfaces/orderData";
+import _ from "lodash";
+
+const Orders = () => {
+  const [page, setPage] = useState<number>(1);
+  const [offset, setOffSet] = useState<number>(OFFSET);
+  const limit = LIMIT;
+  const { data: orders, isLoading, isFetching, isError } = useGetAllOrdersQuery({ limit, offset });
+  const [filterdOrders, setFilterdOrders] = useState<IOrder>();
+  const [filterStatus, setFilterStatus] = useState<FilterType>();
+
+  const totalPages = Math.ceil((orders?.total || []) / limit);
+
+  // deboubce set search bar value
+  const debouncedChangeHandler = useMemo(
```

Comment:
You already wrapped the debounce by useCallback, why need to have a useMemo here? 
Moreover, the filter here seems to have a bug, the search bar does not work for me. In my test, you will get nothing after inputing anything in the search bar.

Reply:
deleted

Reply:
deleted